vignettes/using_lintr.Rmd

```diff
@@ -0,0 +1,320 @@
+---
+title: "Using lintr"
+author: "Alexander Rosenstock"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Using lintr}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+This vignette describes how to set up and configure `lintr` for use with projects or packages.
+
+## Running `lintr` on a project
+
+Checking an R project for lints can be done with three different functions:
+
+* Lint a single file using `lint()`:
+
+```r
+lint(filename = "R/bad.R")
+```
+
+* Lint a directory using `lint_dir()`:
+
+```r
+lint_dir(path = "R")
+```
+
+This will apply `lint()` to all R source files matching the `pattern` argument.
+By default, this means all `.R` files as well as `knitr` formats (e.g. `.Rmd`, `.Rnw`).
+
+`lint_dir` is vectorized over `path`, so multiple directories can be linted at the same time.
+
+* Lint all relevant directories of an R package using `lint_package()`:
+
+```r
+lint_package(path = ".")
+```
+
+This will apply `lint_dir()` to all subdirectories usually containing R code in packages:
+
+ - `R` containing the package implementation.
+ - `tests` containing test code.
+ - `inst` containing sample code or vignettes that will be installed along with the package.
+ - `vignettes` containing package vignettes.
+ - `data-raw` containing code to produce `data` files.
+
+For more information about the assumed package structure, see
+[R Packages](https://r-pkgs.org/package-structure-state.html).
+
+## Configuring linters
+
+### The `.lintr` file
+
+For R projects or R packages the most convenient method of configuration is creating a `.lintr` file in the project
+root. This file is in debian control format (`?read.dcf`) and each key is evaluated as R code by `lintr` when reading
+the settings.
+
+The settings file is searched for in the following order. The first one found, if any, will be used.
+
+ 1. If `options(lintr.linter_file)` is an absolute path, this file will be used.
+    The default for this option is `".lintr"`.
+ 2. A linter file (that is a file named like `lintr.linter_file`) in the currently searched directory,
+    i.e. the directory of the file passed to `lint()`. When ran from `lint_package()`, this directory can vary by each
+    linted file.
+ 3. A linter file in a parent directory of the currently searched directory, starting from the
+    deepest path, moving upwards one level at a time.
+ 4. A linter file in the users `HOME` directory.
+
+If no linter file is found, only default settings take effect (see [defaults](#defaults)).
+
+### Using `options()`
+
+Values in `options()`, if they are not `NULL`, take precedence over those in the linter file (e.g. `.lintr`).
+Note that the key `option_name` in the linter file translates to an R option `lintr.option_name`.
+For example, `options(lintr.exclude = "# skip lint")` will take precedence over `exclude: # nolint` in the linter file.
```

Comment:
Does timing matter for this? e.g. if I run

```
options(lintr.exclude = "# skip lint")
library(lintr)
```

is it any different from the reverse?

Reply:
It doesn't. All settings are read within `read_settings()` when calling a linting function.